### PR TITLE
Add release checklist item to check NOTICE file

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Add entry to check NOTICE file template jdk source reprository list.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Everyone participating in a release, including the release champion are requeste
 - [ ] **Release Champion named** whose responsibility is to ensure every item in this checklist gets completed
 - [ ] **Release Checklist Created**  Create this issue to track the release and the preparation tasks.
 - [ ] **Identify Expected Release Versions** - Find out the version numbers from [here](https://www.java.com/releases/)
+- [ ] **Check/update temurin-build NOTICE file template source repository list** (https://github.com/adoptium/temurin-build/blob/5c7ebf540e2ee298392acc362f0a1f705a0695a2/sbin/NOTICE.template#L41)
```
Comment (zdtsw) [inline .github/ISSUE_TEMPLATE/release-checklist.md:31]: does it mean we need to update this list when a new release come or old EOL?

Comment (andrew-m-leonard) [reply .github/ISSUE_TEMPLATE/release-checklist.md:31]: I believe so

**Comment by tellison** [comment]
IMHO I don't think we should burden the release checklist with this additional item. The NOTICE file does guide people to the source repositories, and shows our 'naming pattern', so I believe people can figure it out themselves.

Updating the repo names on an ad hoc basis should be fine.

**Comment by andrew-m-leonard** [comment]
Thanks @tellison , Yes agree would be nice to keep Checklist smaller

**Pull request closed without merging**